### PR TITLE
Upgrade various things in Build.yaml

### PR DESCRIPTION
**Pull request by JacquesCarette**
let's see if this helps with the timeouts.

### CODE REVIEW

File: .github/workflows/Build.yaml [hunk 1]
```diff
@@ -19,17 +19,17 @@ jobs:
     runs-on: ubuntu-latest
```
Comment (balacij) [inline .github/workflows/Build.yaml:19]: By any chance, can we just revert this back to 20.04?

I'm not sure if Runner upgrades are persistent, but, if they're not, then we might be using up a lot more CI minutes.

Comment (JacquesCarette) [reply .github/workflows/Build.yaml:19]: That's possibly what I will try next.

**Comment by JacquesCarette** [comment]
So the other PR, when restarted, worked. So maybe it's best to not merge this PR and only make the minimal changes?

**Comment by balacij** [comment]
Wow, that's great! Any idea what finally fixed it? In any case, we should for sure update the checkout and cancel-workflow-action tools, so we can at least keep those.

**Review by balacij — APPROVED** [review]
As the changes are now, I like them and I think they should be merged. It also looks like the CI worked well.

I will let you analyze the CI as you mentioned as well, so I will just leave "Approval" without merging.

**Comment by JacquesCarette** [comment]
So the weird part is that it worked, and now it's failing again, somewhere else. I'm guessing something else has updated itself, and now we're encountering a problem. Sigh. There's a new PR to revert things back a bit, for now. I'll open an issue for the test.

**Pull request merged**